### PR TITLE
Refactor project to be more inline with Jetbrains intellij platform plugin template

### PR DESCRIPTION
**Pull request by KotlinIsland**
Refactor a lot of the build / workflow to match: https://github.com/JetBrains/intellij-platform-plugin-template

- Updated Gradle to 7.4
- Convert build.gradle to Kotlin
- Updated Kotlin to 1.6.10
- Simplified dependencies:
  - Removed kotlin stdlib
  - Migrated junit.jupiter to kotlin.test
  - Removed jetbrains.annotations
- Adopted Qodana and integrated it into the CI
- Adopted the changelog Gradle plugin and integrated it into the CI
- Add support for 2022.1

These changes simplify the sources of data within the project:
- Extracting the high level config from the build script / plugin.xml simplifies making changes.
- Moving the changelog into `CHANGLOG.md` makes it more accessible.

I also updated some issues I found in the markdown documents:
- Minor spelling mistakes
- Updated links
- Removing `$` from sample commands so that they are copypaste-able and runnable from within the IDE.

And convert all the Java to Kotlin.

closes: #434
closes: #433
closes: #429
closes: #428

Github sucks and says that I deleted some files when I was actually very careful to rename them to preserve the history.

I'm sure I broke some things 😬. But I hope you like this change :)

### CODE REVIEW

**Comment by KotlinIsland** [comment]
I noticed that #430 will cause a lot of conflicts, do you want me to include 2022.1 support in this pr as well?

**Comment by koxudaxi** [comment]
@KotlinIsland Thank you for creating the cool PR. 😍
Did you know why LGTM analyses are failed?

> I noticed that https://github.com/koxudaxi/pydantic-pycharm-plugin/pull/430 will cause a lot of conflicts, do you want me to include 2022.1 support in this pr as well?

If we can support 2022.1 and 2021.3 then I want to include 2022.1.
Can we do it?

**Comment by KotlinIsland** [comment]
@koxudaxi Awesome that you like the sound of this pr!

I'm not sure what LTGM is, I've never used it before, but it looks like it is also failing in #430

**Comment by KotlinIsland** [comment]
>If we can support 2022.1 and 2021.3 then I want to include 2022.1.

I think so, the plugin verify step passed for 213 and 221
https://github.com/koxudaxi/pydantic-pycharm-plugin/runs/5112235486?check_suite_focus=true#step:10:65

File: resources/META-INF/plugin.xml
Comment (KotlinIsland) [inline resources/META-INF/plugin.xml:1]: @koxudaxi Does the plugin need restart? The plugin verify step said that it could be installed without restart

https://github.com/koxudaxi/pydantic-pycharm-plugin/runs/5112235486?check_suite_focus=true#step:10:72
> Plugin can be loaded/unloaded without IDE restart

File: build.gradle.kts
Comment (KotlinIsland) [inline build.gradle.kts:28]: Previously this was `pluginName = project.name`(pydantic-pycharm-plugin), but this change makes it "Pydantic", I'm not sure what you would want here.

File: build.gradle.kts
Comment (KotlinIsland) [inline build.gradle.kts:104]: I can't see any of this configuration from before this merge, maybe this section should be removed?

**Comment by koxudaxi** [comment]
Thank you for your changes.
I'm checking the commits!!

File: resources/META-INF/plugin.xml
Comment (koxudaxi) [reply resources/META-INF/plugin.xml:1]: @KotlinIsland 
I forget why I add the restart option 🤔 
If we don't need the option then we should remove it.

**Comment by KotlinIsland** [comment]
> I don't know about `Qodana` . The tools can check kotlin's code?

[Qodan](https://www.jetbrains.com/qodana/) is a code quality tool developed by JetBrains that supports Kotlin, It's recommended by the Jetbrains IntelliJ platform plugin template, and is basically the official Kotlin code quality tool.

**Comment by koxudaxi** [comment]
@KotlinIsland 

> It has Gradle support, CLI support, Ide support and A dedicated Github App.
Great!!

> It has Gradle support, CLI support, Ide support and A dedicated Github App.
Could you please set up Qodana? And remove LGTM in GitHub Action?

File: resources/META-INF/plugin.xml
Comment (KotlinIsland) [reply resources/META-INF/plugin.xml]: Seems you changed it with message "Fix build warning", I've removed it now and will see if there are any build warnings.

**Comment by KotlinIsland** [comment]
@koxudaxi 

I have added Qodana to the CI. I am unable to remove LGTM, I think it is a project setting.

If you would like to add the Github Qodana App(Which I recommend) look [here](https://github.com/apps/qodana/) to sign up and register this project.

File: build.gradle.kts
Comment (koxudaxi) [reply build.gradle.kts:28]: Me too, I forget about the code😅
We should remove the code.

**Comment by koxudaxi** [comment]
@KotlinIsland 
I have deactivated LGTM and set up  Qodana App!!

**Review by koxudaxi — APPROVED** [review]
@KotlinIsland 
LGTM
Can I merge the PR?

**Comment by koxudaxi** [comment]
@KotlinIsland 
I can't use the service?
Should I send an email to the address?
スクリーンショット 2022-02-13 1 38 52

**Comment by KotlinIsland** [comment]
Yeah, you need to send an email at the moment

https://www.jetbrains.com/help/qodana/qodana-github-application.html#how-to-start-github-app

**Comment by koxudaxi** [comment]
@KotlinIsland 
I joined the group. But, the tools didn't run. :(
https://github.com/koxudaxi/pydantic-pycharm-plugin/pull/440

**Comment by KotlinIsland** [comment]
@koxudaxi
Did you get a response email from JetBrains yet? If you have then maybe send them another email with follow up explaining that it's not working. Strangely my projects have stopped working with the Qodana app today as well.

Qodana is currently integrated into the [`build` workflow](https://github.com/koxudaxi/pydantic-pycharm-plugin/runs/5185929196?check_suite_focus=true#step:12:106), and you can download the report from the workflow [here](https://github.com/koxudaxi/pydantic-pycharm-plugin/actions/runs/1841570305)

**Comment by koxudaxi** [comment]
@KotlinIsland 
I found the message from the support team in the mailbox.

_I suggest you also try [Qodana Scan](https://github.com/marketplace/actions/qodana-scan). It's a GitHub action with more flexible settings and abilities than the app. And it's likely that we'll drop support of the app in the near future._

**Comment by koxudaxi** [comment]
@KotlinIsland 
Thank you for your great job.
I have released the version.
But, I find an error in the action.
Do you know the reason for the error?
https://github.com/koxudaxi/pydantic-pycharm-plugin/runs/5290676713?check_suite_focus=true
Screenshot 2022-02-23 at 0 45 00

**Comment by KotlinIsland** [comment]
@koxudaxi I know the issue, I'll make a pr soon, it's related to the branch names.

**Pull request merged**
